pkg/util/kubernetes/apiserver/metadata_controller_test.go

```diff
@@ -430,15 +432,25 @@ func TestMetadataController(t *testing.T) {
 	informerFactory.Start(stop)
 	go metaController.Run(stop)
 
-	for !metaController.endpointsListerSynced() && !metaController.nodeListerSynced() {
-		time.Sleep(150 * time.Millisecond)
-	}
+	retry.DoWithRetry(t, "wait Lister Synced", 10, 50*time.Millisecond, func() (string, error) {
```

Comment:
interesting, this is very similar to our own testutil here: https://github.com/DataDog/datadog-agent/blob/7f5ebddd1a3d36e705c59262ee02faf2ecf1a2b4/pkg/util/testutil/timeout.go#L29

It'd be good to use one or the other across the test codebase, but I guess we can address this later. `gruntwork-io/terratest` looks interesting, there are other test utils there that could prove useful for other tests.

Reply:
I didn't know the `assertTrueBeforeTimeout` I will use it.

Reply:
done